### DISCUSSION TITLE
Docker: Multiple Certificates per Container?

### DISCUSSION THREAD

**Discussion by DaCHack** [go-acme/lego#2042] [Q&A]
Hi,

can I run a lego container and make it request 2 separate certificates (under the same email address or different ones does not matter) or will I need 2 separate containers with their own volume each?
If so, how to configure the docker-compose.yaml accordingly?

Many thanks!

**Comment by ldez** [comment]
Hi,

what do you call "separate certificates"?

**Reply by DaCHack** [reply]
Let's say I have 2 machines in the network (server1.me.tld and server2.me.tld).
Currently, my lego container retrieves a single certificate for server1.me.tld with server2.me.tld in the alternative name. THis cert I copy over to server2.
But going forward I only want server2 to have a valid certificate for its original domain and not for the others in the network as well.

So "separate" would mean for me that I actually retrieve 2 certificates and not one with further domain names in the alternative names.

Does this answer your question?

**Reply by ldez** [reply]
You just have to call 2 times lego: one with a domain, one with another.
You can use the same volume mount, but you have to run those commands sequentially to avoid concurrency on the account file.

**Reply by DaCHack** [reply]
Ah okay, thanks for clarifying. This is going to be the issue. I let the container run continuously to update the cert automtically.

Thus, I would Need 2 volumes as well or risk concurrency issues. No chance to cover this case for elegant through command line options and include Everything into one Container?

thanks!

**Reply by ldez** [reply]
Technically, the currency problem can only happen when the certificates are created for the first time because the account file is created at this time.

If you continuously run, I think you are using 2 crons and 2 calls, so you can either configure your crons to start not at the same time or just add 2 different waits before the run.

**Comment by DaCHack** [comment]
Or let me put it different:
Can I run multiple Lego containers for different certs on the same bind mounted volume without letting them break wach other?
This way I Could have one container keeping server1 certs up to Date and another one for server2
